### PR TITLE
Release 0.25.2

### PR DESCRIPTION
**Pull request by otaviopace**

### CODE REVIEW

File: core/src/subgraph/instance_manager.rs [hunk 1]
```diff
@@ -42,6 +44,14 @@ lazy_static! {
...
     pub static ref DISABLE_FAIL_FAST: bool =
         std::env::var("GRAPH_DISABLE_FAIL_FAST").is_ok();
+
+    /// Ceiling for the backoff retry of non-deterministic errors, in seconds.
+    pub static ref SUBGRAPH_ERROR_RETRY_CEIL_SECS: Duration =
+        std::env::var("GRAPH_SUBGRAPH_ERROR_RETRY_CEIL_SECS")
+            .unwrap_or((MINUTE * 30).as_secs().to_string())
+            .parse::<u64>()
+            .map(Duration::from_secs)
+            .expect("invalid GRAPH_SUBGRAPH_ERROR_RETRY_CEIL_SECS");
```
Comment (mangas) [inline core/src/subgraph/instance_manager.rs:54]: nitpick: I think MAX_SECS would prolly be an improvement from CEIL

File: core/src/subgraph/instance_manager.rs [hunk 2]
```diff
@@ -458,9 +468,30 @@ where
...
+    // deterministic.
+    if let Some(current_ptr) = ctx.inputs.store.block_ptr()? {
+        if let Some(parent_ptr) = ctx.inputs.triggers_adapter.parent_ptr(&current_ptr).await? {
+            // This reverts the deployment head to the parent_ptr if
+            // deterministic errors happened.
+            //
+            // There's no point in calling it if we have no current or parent block
+            // pointers, because there would be: no block to revert to or to search
+            // errors from (first execution).
+            let _outcome = ctx
```
Comment (mangas) [inline core/src/subgraph/instance_manager.rs:484]: do we need to capture _outcome here at all?

File: core/src/subgraph/instance_manager.rs [hunk 3]
```diff
@@ -686,21 +697,70 @@ where
...
+                                .remove(&ctx.inputs.deployment.id);
+
+                            error!(logger, "Subgraph failed for non-deterministic error: {}", e;
+                                "attempt" => backoff.attempt,
+                                "retry_delay_s" => backoff.delay().as_secs());
+
+                            // Sleep before restarting.
+                            backoff.sleep_async().await;
+
+                            should_try_unfail_non_deterministic = true;
```
Comment (mangas) [inline core/src/subgraph/instance_manager.rs:758]: these boolean flags are super easy to miss, feels like we could refactor a lot of these into a state machine. It's probably not something to change now but to keep in mind

**Review by leoyvens — APPROVED** [review]

**Comment by otaviopace** [comment]
This can be closed, just got released as git tag / github release: https://github.com/graphprotocol/graph-node/releases/tag/v0.25.2

**Pull request closed without merging**